subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectInternal.java

```diff
@@ -101,6 +102,20 @@ public interface ProjectInternal extends Project, ProjectIdentifier, HasScriptSe
 
     void subprojects(ProjectInternal referrer, Action<? super Project> configureAction);
 
+    /**
+     * Do not use this method to access the child projects in the Gradle codebase!
+     * The implementations may add check that enforce correct usage of the public API, such as
+     * cross-project model access checks, which are meant to report warnings on incorrect API usages
+     * from third-party code. The internal usages won't pass these checks and will break.
+     *
+     * @see ProjectInternal#getChildProjectsInternal()
+     * @see ProjectHierarchyUtils#getChildProjectsForInternalUse(Project)
+     */
+    @Override
+    Map<String, Project> getChildProjects();
+
+    Map<String, Project> getChildProjectsInternal();
```

Comment:
🎨 I'd probably choose a more specific suffix, something like `Unchecked` or `Unsafe` instead of `Internal`.

Reply:
Yep, `Unchecked` sounds good to me.